wdl/transforms/draft2/src/main/scala/wdl/transforms/draft2/wdlom2wom/WdlDraft2WomWorkflowDefinitionMaker.scala

```diff
@@ -0,0 +1,25 @@
+package wdl.transforms.draft2.wdlom2wom
+
+import common.validation.ErrorOr.ErrorOr
+import wdl.WdlWorkflow
+import wom.callable.{WomWorkflowDefinitionMaker, WorkflowDefinition}
+
+object WdlDraft2WomWorkflowDefinitionMaker extends WomWorkflowDefinitionMaker[WdlWorkflow] {
+  override def toWomWorkflowDefinition(wdlWorkflow: WdlWorkflow): ErrorOr[WorkflowDefinition] = {
+    // NB: We don't allow "OuterGraphInputNode"s when building this (the Map is empty), so preserveScatterForExternalLookups isn't ever actually used.
+
+    WdlDraft2WomGraphMaker.toWomGraph(wdlWorkflow, Set.empty, Map.empty, preserveIndexForOuterLookups = true) map { wg =>
+      WorkflowDefinition(
+        wdlWorkflow.unqualifiedName,
+        wg,
+        wdlWorkflow.meta,
+        wdlWorkflow.parameterMeta,
+        List.empty)
+    }
+  }
+
+  // TODO Dan: Is adding this back as an implicit class a hideous abuse of power, or the whole point of doing this?
+  implicit class Draft2EnhancedWdlWorkflow(val wf: WdlWorkflow) extends AnyVal {
+    def toWomWorkflowDefinition = WdlDraft2WomWorkflowDefinitionMaker.toWomWorkflowDefinition(wf)
```

Comment:
@danbills, I had a thought after I made the PR. I suspect the magic I've missed here is along the lines of giving this implicit class to the "WomXMaker" type classes rather than the specializations, something like (not tried yet, just want to make sure I'm not going down the wrong path):
```
trait XMaker[A] { def makeX(a: A) }
implicit class xMakerWrapper[A](a: A)(implicit val xMakerInstance: XMaker[A]) {
  def turnIntoAnX = xMakerInstance(a)
}
```

Reply:
This is absolutely the correct path, and it's so common that there's a library to generate this boilerplate for you!  Even better, it's already in scope.  

Use the `@typeclass` annotation before `trait XMaker[A]`, and it gives you that `XMakerWrapper` for free.  See https://github.com/mpilquist/simulacrum for the details and how to import (creates a .ops that must be imported to use the generated stuff)

Reply:
Bleargh. I just went away and added these manually. But I learned a lot by doing it and shifting the typeclasses out into their own package was a good idea regardless so not totally wasted effort.